### DISCUSSION TITLE
x-id and $id are not updated correctly

### DISCUSSION THREAD

**Discussion by alterastro** [alpinejs/alpine#2844] [7. Bugs (archived)]
Hello,
There is a component structure:
```
<div x-data="{modelValue: 'test-id', id: 'test', setId() { return this.id = this.modelValue }}" x-id="[id]">
  <label :for="$id(id)">Test label</label>

  <input x-model="modelValue" @input.stop="setId" :id="$id(id)" type="text" />
</div>
```
During initialization, exactly what we expect according to the description in the documentation happens:
```
<div x-data="{modelValue: 'test-id', id: 'test', setId() { return this.id = this.modelValue }}" x-id="[id]">
  <label :for="$id(id)" for="test-1">Test label</label>

  <input x-model="modelValue" @input.stop="setId" :id="$id(id)" type="text" id="test-1">
</div>
```
But after changing the value of the id property, x-id and $id stop working as expected:
```
<div x-data="{modelValue: 'test-id', id: 'test', setId() { return this.id = this.modelValue }}" x-id="[id]">
  <label :for="$id(id)" for="test-idx-1">Test label</label>

  <input x-model="modelValue" @input.stop="setId" :id="$id(id)" type="text" id="test-idx-2">
</div>
```
If you write the same example without using x-id, then the update will occur correctly:
```
<div x-data="{modelValue: 'test-id', id: $id('test'), setId() { return this.id = `${$id(this.modelValue)}` }}" x-id="[id]">
  <label :for="id">Test label</label>

  <input x-model="modelValue" @input.stop="setId" :id="id" type="text" />
</div>
```
As a result, when changing the id, we will get the following result:
```
<div x-data="{modelValue: 'test-id', id: $id('test'), setId() { return this.id = `${$id(this.modelValue)}` }}" x-id="[id]">
  <label :for="id" for="test-idx-1">Test label</label>

  <input x-model="modelValue" @input.stop="setId" :id="id" type="text" id="test-idx-1">
</div>
```
Either I don’t understand something in the x-id mechanism, or there is an error here.

**Comment by SimoTod** [comment]
Looking at the source code, it seems that x-id is not a reactive property so it's not expected to change after the component initialisation.

**Reply by alterastro** [reply]
Thanks for your reply. I didn’t think to get into the source right away)
Does it make sense to request a feature with a reactive id, or does it go against Alpine's logic?
I have several cases where this would be useful.

**Reply by SimoTod** [reply]
It don't think it goes against Alpine but it's probably easier with a PR than a feature request.
